lib/media/segment_index.js

```diff
@@ -700,6 +700,24 @@ shaka.media.MetaSegmentIndex = class extends shaka.media.SegmentIndex {
     this.indexes_ = [];
   }
 
+  /** Evicts all old SegmentIndexes in this MetaSegmentIndex that are empty. */
+  evictEmpty() {
+    if (this.indexes_.length == 0) {
+      return;
+    }
+    const prevNewestIndex = this.indexes_[this.indexes_.length - 1];
+    while (this.indexes_.length > 0 && this.indexes_[0].isEmpty()) {
```

Comment:
```suggestion
    while (this.indexes_.length > 1 && this.indexes_[0].isEmpty()) {
```
With this we could get rid of making an explicit assertion and avoid issues.